pandas/core/indexes/base.py

```diff
@@ -1339,6 +1342,237 @@ def rename(self, name, inplace=False):
         """
         return self.set_names([name], inplace=inplace)
 
+    def replace(
```

Comment:
Can you add type annotations

Reply:
Won't do: https://github.com/pandas-dev/pandas/pull/32542#issuecomment-646993103